program/src/duplicate_block_proof.rs

```diff
@@ -862,11 +1106,78 @@ mod tests {
             .iter()
             .flat_map(|(a, b, c)| [(a, b, c), (b, a, c)])
         {
-            let proof_data = generate_proof_data(shred1, shred2);
+            let (proof_data, context) = generate_proof_data(&leader_pubkey, shred1, shred2);
             assert_eq!(
-                proof_data.verify_proof(SLOT, &leader.pubkey()).unwrap_err(),
+                proof_data
+                    .verify_proof(context, SLOT, &leader_pubkey)
+                    .unwrap_err(),
                 *expected,
             );
         }
     }
+
+    #[test]
+    fn test_sigverify() {
+        let mut rng = rand::thread_rng();
+        let leader = Arc::new(Keypair::new());
+        let leader_pubkey = leader.pubkey();
+        let shredder = Shredder::new(SLOT, PARENT_SLOT, REFERENCE_TICK, VERSION).unwrap();
+        let next_shred_index = rng.gen_range(0..32_000);
+        let shred1 =
+            new_rand_data_shred(&mut rng, next_shred_index, &shredder, &leader, true, true);
+        let shred2 =
+            new_rand_data_shred(&mut rng, next_shred_index, &shredder, &leader, true, true);
+
+        let (proof_data, context) = generate_proof_data(&leader_pubkey, &shred1, &shred2);
+        assert!(proof_data
+            .verify_proof(context, SLOT, &leader_pubkey)
+            .is_ok());
```

Comment:
nit: I typically prefer doing an unwrap() to get a better error message if something goes wrong, ie
```suggestion
        proof_data
            .verify_proof(context, SLOT, &leader_pubkey)
            .unwrap();
```